src/app/shared/main-layout/header/header.component.ts

```diff
@@ -1,13 +1,42 @@
-import { Component } from "@angular/core";
+import { AfterContentChecked, Component } from "@angular/core";
+import { AuthService } from "../../../core/index";
+import { Router } from "@angular/router";
 
 @Component({
     selector: "app-header",
     templateUrl: "./header.component.html",
     styleUrls: ["./header.component.scss"]
 })
-export class HeaderComponent {
+export class HeaderComponent implements AfterContentChecked {
     /**
      * placeholder for template.
      */
-    public user = "User login";
+    public user: string;
+
+    private authService: AuthService;
+    private router: Router;
+
+    /**
+     * method that calls logout method of AuthService
+     */
+    public logoff(): void {
+        this.authService.logout();
+        this.router.navigate(["/login"]);
+    }
+
+    constructor(authService: AuthService, router: Router) {
+        this.authService = authService;
+        this.router = router;
+
+    }
+
+    public checkStatus(): boolean {
+        return this.authService.isAuthenticated();
+    }
+
+    public ngAfterContentChecked() {
```

Comment:
something goes wrong if you are using afterContentChecked

this method is called frequently and may become a CPU leak in your application.
It would be better to organize auth service with observables.

Reply:
fixed